src/nvim/terminal.c

```diff
@@ -793,6 +794,57 @@ static VTermKey convert_key(int key, VTermModifier *statep)
     case K_KMULTIPLY: return VTERM_KEY_KP_MULT;
     case K_KDIVIDE:   return VTERM_KEY_KP_DIVIDE;
 
+    case K_S_F1:      *statep |= VTERM_MOD_SHIFT;
```

Comment:
Good point, but I think that's still preferable: 

- the logic is a single line instead of repeated
- modification of `statep` continues to be driven by `convert_modifiers`